src/ClickableAttributes.elm

```diff
@@ -102,12 +112,24 @@ toButtonAttributes clickableAttributes =
 
 
 {-| -}
-toLinkAttributes : ClickableAttributes msg -> ( String, List (Attribute msg) )
-toLinkAttributes clickableAttributes =
+toLinkAttributes : (route -> String) -> ClickableAttributes route msg -> ( String, List (Attribute msg) )
+toLinkAttributes routeToString clickableAttributes =
```

Comment:
🎨  It feels like there are some implicit rules here about what the valid combinations of `url`/`urlString` and `linkType` are, and I wonder if modelling those would simplify this function and make this easier to understand overall.

I'm thinking something like this:

```elm
type alias ClickableAttributes route msg =
    { linkType : Link route
    , onClick : Maybe msg
    }


type Link route
    = Default
    | WithTracking route
    | SinglePageApp route
    | WithMethod String route
    | External String
    | ExternalWithTracking String
```

It seems like:
  - `Default` is the case in which there isn't any target `url`/`urlString`
  - We only use `urlString` with external links
  - For all other cases we use `url`.

If this made sense, we could implement this with a single case, grabbing the target link from the `Link`'s constructor params.

Reply:
Thanks for the suggestion here! I'm going to link this in an issue, and come back to it when I've got a little more bandwidth. 🙏